app/main.py

```diff
@@ -1,3 +1,23 @@
-def cache(func):
-    # Write your code here
-    pass
+from functools import wraps
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    caching_result = []
```

Comment:
Use `dict` for caching results, you could store `args` as key and calculation result as value. `dict` is a more preferred data structure here.